hyperspy/_signals/signal1d.py

```diff
@@ -1546,6 +1548,30 @@ def estimating_function(spectrum,
 
     estimate_peak_width.__doc__ %= (SHOW_PROGRESSBAR_ARG, PARALLEL_ARG, MAX_WORKERS_ARG)
 
+    def plot(self,
+             navigator="auto",
+             plot_markers=True,
+             intensity_autoscale=True,
+             norm="auto",
+             axes_autoscale=False,
```

Comment:
In the 1D case I would rename:

* ``axes_autoscale`` -> ``xaxis_autoscale``
* ``intensity_autoscale`` -> ``yaxis_autoscale``

Reply:
I still prefer the current syntax because it is consistent between 1D and 2D signal. More importantly, the `xaxis` and `yaxis` is consistent with matplotlib but not hyperspy: `x` and `y` in the context of axis in hyperspy can refer to navigation axis, which is then misleading!

Reply:
The issue is that ``axes_autoscale`` is plural, while it just affects the x-axis, what I find confusing. As for ``intensity_autoscale``, it is not obvious that it refers to the ``y-axis``. I don't think that consistency with the 2D counterparts is required or even desired, since they actually act pretty differently. For examples ``axes_autoscale`` affects both axes in 2D but just the x axis in 1D. Furthermore, the docstring resorts to the x/y nomenclature to clarify which axes the keywords operate on...

Reply:
Good points, but I still think it would convenient to be consistent between 1D and 2D figure (actually not signal) because it can apply to the navigator too and be used through the `navigator_kwds` argument. These arguments will do different things in case of 1D and 2D but conceptually, there are equivalent.

Clearly, the docstring needs improvement! ;)

Reply:
I think that that's one more argument for using different keywords because people tend to think that things with the same name do the same thing...

Reply:
> I think that that's one more argument for using different keywords because people tend to think that things with the same name do the same thing...

This doesn't mean that it is a good reason! Thinking about it a bit, I am convincing myself that it should be:

* `axes_autoscale` -> `axis_autoscale`
* `intensity_autoscale` stay the same

Because in hyperspy, we use the term `intensity` (e.g. in the `norm` argument) and here `axis` it refers to an `axis` of the `axes_manager`, and if there are 1, 2 or navigation or signal axes, it should not matter. 
The only occurrence of `xaxis` is in `xaxis_type` for plotting explained variance:
https://github.com/hyperspy/hyperspy/blob/54856fd9c9aefb62f7ea68f1d90431072151d060/hyperspy/learn/mva.py#L1294-L1307

Reply:
I don't like ``axis_autoscale`` because it is not precise enough, which axis is it referring to?

I have another suggestion to try to break the impasse: use simply one keyword that can take `"x"`, `"y"` or `"xy"`. What do you think?

Reply:
> I don't like `axis_autoscale` because it is not precise enough, which axis is it referring to?

Well, it is referring to the relevant `axis` from the `axes_manager` (depending, if this is a navigator or signal Figure) against which the data are displayed.

> I have another suggestion to try to break the impasse: use simply one keyword that can take `"x"`, `"y"` or `"xy"`. What do you think?

I have been this route before and I find it confusing... I would actually prefer your first suggestion.

Reply:
OK, I think that you'll like this small variation of my previous suggestion because it helps with consistency that is what you are (rightly) after: just one keyword that could be ``autoscale`` or ``axes_autoscale`` that takes `"x"`, `"y"` or `"xy"` (`"x"` and `"y"` not yet implemented for 2D. If we choose simply ``"autoscale"`` (my favourite), then we could remove the 2D ``intensity_autoscale`` keyword, since for 2D it could take `"x"`, `"y"`, `"z"` (for intensity) and any combination of those. I like this one, because it is minimalist, consistent and intuitive.

Reply:
Yes, this is good. Previously, I have considered `x`, `y`, or `xy` (but not `xyz`) and using `intensity_autoscale` separately, which was not great but having a combination of all of them in the same argument (`autoscale`) is nice - as you said minimalist and intuitive.

I can work on it this evening but if someone want to take over earlier, that's more than welcome! :)

Reply:
After thinking on it, maybe it is not actually a good idea to include `"z"` in ``autoscale`` because it is redundant with (and less rich than) #2272's `"th"` syntax. So, maybe just ``axes_autoscale`` that behaves similarly (exactly the same is not possible) for 1D and 2D would do?

Reply:
This should be fine if we disable *z autoscaling* when `vmin` and/or `vmax` are provided?

Reply:
We could indeed make `"z"` autoscaling a shortcut for ``vmin="0th", vmax="100th"`` but then, what happens if ``"z"`` is not in ``autoscale`` and ``vmin`` and ``vmax`` are not provided? I think that, when in doubt, simpler is better. Once ``saturated_pixels`` is gone, we can make by default ``vmin="0th", vmax="100th"``. In this case, disabling z-autoscaling simply involves setting a numeric value, what requires the user to know what he's doing.

Reply:
As it is currently implemented, if `vmin` and `vmin` are not provided and `'z'` is not in `autoscale`, the contrast will be set with for the first image to display. If the data change or the indices of the axes manager changes, the range of displayed values will not changed.
For hyperspy 2.0, when `saturated_pixels` is removed, then we can set the default values `vmin="0th"` and `vmin="100th"` and remove allowing setting `None` to `vmin`, `vmax` for simplification.

The alternative is to raise an error when `'z'` is not in `autoscale` and `vmin`/`vmax` are not provided, which is possibly what you mentioned earlier?